ops/catalog/migrations/0001_create_catalog.sql

```
@@ -0,0 +1,52 @@
+  CONSTRAINT package_versions_cas_index_id_check CHECK (
+    cas_index_id IS NULL OR length(btrim(cas_index_id)) > 0
+  ),
+  CONSTRAINT package_versions_error_check CHECK (
+    (state = 'FAILED' AND error IS NOT NULL AND length(btrim(error)) > 0)
+    OR (state <> 'FAILED' AND error IS NULL)
+  )
+);
+
+CREATE INDEX package_versions_reconcile_eligible_idx
+  ON package_versions (attempts, next_attempt_at, updated_at, id)
+  WHERE state IN ('UPLOADING', 'ASSEMBLED', 'PROMOTING', 'FAILED');
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)